presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConfigAPI.java

```diff
@@ -35,40 +36,66 @@
 
 @Path("/config")
 @Api(description = "the config API")
-@javax.annotation.Generated(
-        value = "io.swagger.codegen.languages.JavaJAXRSSpecServerCodegen",
-        date = "2017-06-23T09:53:13.549-04:00")
 @Singleton
 public final class ConfigAPI
 {
     private static final APIFileHandler apiFileHandler = new APIFileHandler(Paths.get("/etc/presto"));
 
     @GET
-    @Produces({MediaType.TEXT_PLAIN})
+    @Produces(MediaType.TEXT_PLAIN)
     @ApiOperation(value = "Get available configuration files")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "Retrieved configuration", response = String.class)})
+            @ApiResponse(code = 200, message = "Retrieved configuration")})
     public synchronized Response getConfig()
     {
         return apiFileHandler.getFileNameList();
     }
 
     @GET
     @Path("/{file}")
-    @Produces({MediaType.TEXT_PLAIN})
+    @Produces(MediaType.TEXT_PLAIN)
     @ApiOperation(value = "Get configuration by file")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "Retrieved file", response = String.class),
+            @ApiResponse(code = 200, message = "Retrieved file"),
             @ApiResponse(code = 404, message = "Resource not found")})
     public synchronized Response getConfigFile(
             @PathParam("file") @ApiParam("The name of a file") String file)
     {
         return apiFileHandler.getFile(file);
     }
 
+    @GET
+    @Path("/{file}/{property}")
+    @Produces(MediaType.TEXT_PLAIN)
+    @ApiOperation(value = "Get specific configuration property")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Retrieved property"),
+            @ApiResponse(code = 404, message = "Resource not found")})
+    public synchronized Response getConfigProperty(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            @PathParam("property") @ApiParam("A specific property") String property)
+    {
+        return apiFileHandler.getFileProperty(file, property);
+    }
+
+    @PUT
+    @Path("/{file}/{property}")
+    @Produces(MediaType.TEXT_PLAIN)
+    @ApiOperation(value = "Update or create a certain property of configuration file")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Property updated or created"),
+            @ApiResponse(code = 404, message = "Resource not found")})
+    public synchronized Response updateConfigProperty(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            @PathParam("property") @ApiParam("A specific property") String property,
+            String value)
+    {
+        return apiFileHandler.updateProperty(file, property, value);
+    }
+
```

Comment:
Why'd you move `GET /file/property`? I think it would be better to put both of these functions after the operations on `/file`.

Then you wouldn't have to delete the one that's already there, too.